### PR TITLE
allow regex filter for engine name

### PR DESCRIPTION
**Pull request by robertnurnberg**
Allow filtering of pgn data based on engine name. Sample usage: `./scoreWDLstat -o master.json --matchEngine "Base-.*"` to filter just master games and `./scoreWDLstat -o commit-12345.json --matchEngine ".*-12345"` to filter only games for a certain commit.

With some bash script magic, this would allow tracking of SF's WDL model in time.

PS: I also fixed the `Progress` indicator, so that now it starts at 1 and ends with `files_chunked.size()`.

### CODE REVIEW

File: scoreWDLstat.cpp [hunk 1]
```diff
@@ -104,14 +105,34 @@ float fast_stof(const char *str) {
...
 /// @param game
-void ana_game(map_t &pos_map, const std::optional<Game> &game) {
+/// @param regexStr
+void ana_game(map_t &pos_map, const std::optional<Game> &game, const std::string &regexStr) {
     if (game.value().headers().find("Result") == game.value().headers().end()) {
         return;
     }
 
+    Color filterSide = Color::NONE;
+    if (regexStr != "") {
```
Comment (Disservin) [inline scoreWDLstat.cpp:118]: c++ way would be to say
```cpp
if (!regexStr.empty())
```

File: scoreWDLstat.cpp [hunk 2]
```diff
@@ -329,20 +353,21 @@ bool find_argument(const std::vector<std::string> &args,
...
 /// @param argc
-/// @param argv Possible ones are --dir, --file and -o
+/// @param argv Possible ones are --dir, --file, --matchEngine and -o
 /// @return
 int main(int argc, char const *argv[]) {
     const std::vector<std::string> args(argv + 1, argv + argc);
 
     std::vector<std::string> files_pgn;
-    std::string json_filename = "scoreWDLstat.json";
+    std::string regexStr = "", json_filename = "scoreWDLstat.json";
```
Comment (Disservin) [inline scoreWDLstat.cpp:370]: You dont need to assign `""` to a string.

**Review by Disservin — COMMENTED** [review]
current code style is that all variables are in snake case.

**Comment by robertnurnberg** [comment]
Fixed style issues.

**Review by Disservin — APPROVED** [review]
lgtm

**Pull request merged**